erc20-connector/contracts/ProofKeeper.sol

```diff
@@ -0,0 +1,56 @@
+pragma solidity ^0.6.12;
+
+import "rainbow-bridge/contracts/eth/nearprover/contracts/INearProver.sol";
+import "rainbow-bridge/contracts/eth/nearprover/contracts/ProofDecoder.sol";
+import "rainbow-bridge/contracts/eth/nearbridge/contracts/Borsh.sol";
+
+contract ProofKeeper {
```

Comment:
I'm not fully onboard with this new name. I understand your motivation for more generic connector that goes beyond locking tokens, but in this particular case `Locker` makes perfectly clear the goal of this contract while `ProofKeeper` doesn't IMO.

Reply:
Sure, I will split this into separate PRs (though 1 and 3 items will be in one as they are dependent).

What's about `Locker` - I believe that this name is truly misleading because actually, the contract locks nothing. And moreover, it contains just a `_parseProof` method and a map of used events and references the appropriate proof producer. So I don't think that old name doesn't represent the behavior of the contract. 

On the other hand, the map which is inside - `usedEvents` (I believe that could be renamed to `usedProofs` because these are actually proofs, not events) - and `_parseAndConsumeProof` are there and the contract just keeps used proofs and is able to parse and consume proofs meanwhile verifying that this originates from the expected proof producer. And also as we already have `ProofDecoder` this doesn't look foreign IMO. But I agree that `ProofKeeper` isn't the perfect name as well. Perhaps, `ProofConsumer` or `ProofController` are a little bit better names but I'm not sure.  

But if you think that it's better to keep old names for this connector, we can do this way.